internal/rbd/rbd_util.go

```diff
@@ -1333,6 +1334,29 @@ func lookupRBDImageMetadataStash(path string) (rbdImageMetadataStash, error) {
 	return imgMeta, nil
 }
 
+func updateRBDImageMetadataStash(path, device string) error {
+	if device == "" {
+		return fmt.Errorf("device is empty")
+	}
+	imgMeta, err := lookupRBDImageMetadataStash(path)
```

Comment:
I had similar thoughts initially, but thanks to the below code comments, they warned me not to do so:

```
// Stash image details prior to mapping the image (useful during Unstage as it has no   
// voloptions passed to the RPC as per the CSI spec)                                    
err = stashRBDImageMetadata(volOptions, stagingParentPath)                              
if err != nil {                                                                         
    return nil, status.Error(codes.Internal, err.Error())                               
}                                                                                       
```
Given the device name is only known after mapping, this might not be possible.